modules/tensor_mechanics/include/kernels/GeneralizedPlaneStrainOffDiag.h

```diff
@@ -10,6 +10,7 @@
 
 #include "Kernel.h"
 #include "DerivativeMaterialInterface.h"
+#include "ScalarVariableIndexProvider.h"
```

Comment:
I think this is a good way to handle this -- I'm just not sure about the name of this.  Let's talk about this a little bit today.